source/schemas/types.txt

```diff
@@ -509,3 +509,123 @@ with the ``title`` field set to the embedded object's type name:
+- ``type`` (string), which is always set to "Point"
+
+- ``coordinates`` (array of doubles). The ``coordinates`` array must contain  
+  at least 2 double values, and may contain a third:
+    
+  - The first double is the longitude of the point, between -180 and 180.
+  - The second is the latitude, between -90 and 90.
+  - The optional third value represents the elevation/altitude of the point, 
+    in meters. Atlas ignores this value when performing queries. 
+
+To store a GeoPoint in Atlas for later querying with :ref:`Atlas Device Sync <sync>`, 
+the following rules also apply:
+
+- It must be of type ``object``
+- It must be embedded within another type
+- Additional properties are allowed within the ``coordinates`` array, as long as 
+  they are doubles.
+- Additional properties in the GeoPoint object are also allowed.
+ 
+The following is the Device Sync schema of an object with an embedded GeoPoint 
+property named "location":
+
+.. code-block:: json
+   :copyable: false
+
+   {
+     "title": "MyObject",
+     "properties": {
+        "location": {
+          "bsonType": "object",
+          "required": [ "type" ],
+          "properties": {
+            "type": {
+              "bsonType": "string",
+            },
+            "coordinates": {
+              "bsonType": "array",
+              "items": {
+                "bsonType": "double"
+              }
+            }
+          }
+        }
+      }
+   }
+
+The following code block shows a document that follows the schema. It has a 
+single embedded GeoPoint object named "location". Both the containing object 
+and the GeoPoint object contain an extra property.
+
+.. code-block:: json
+   :copyable: false
+   :emphasize-lines: 4-9
+
+   {
+     "_id": { "$oid": "65039d09fe4e46dddee31a3f" },
+     "location": {
+       "type": "Point",
+       "coordinates": [-122.4, 48.12, 23.0],
+       "optional_property": true
+      },
+     "another_optional_property": "geopoint name"
+   }
+
+GeoJSON Points and Device Sync Queries
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+GeoPoints are the only form of geospatial data that Device Sync can 
+query. The Device Sync server validates every embedded GeoPoint property with 
+2 rules:
+
+- The "type" field can only be set to the string "Point" (case-sensitive), 
+- The "coordinates" field can only be set to an array of doubles. The first element 
+  in this array (longitude) must fall in the range of [-180, 180], 
+  and the second element (latitude) must fall in the range of [-90, 90].
+
+If a device writes an object that fails validation, it will be reverted with a 
+:ref:`compensating write error <flexible-sync-errors>`. If a document that doesn't 
+meet these requirements is written directly to the MongoDB collection, the server 
+marks that document as unsyncable.
+
+Querying
+````````
+You can query a GeoPoint using :ref:`Realm Query Language (RQL) <rql>`, 
+using the :manual:`$geoWithin </reference/operator/query/geoWithin>` to do so.
```

Comment:
This shouldn't link to `$geoWithin` in the manual - that's a MongoDB geospatial operator. What we want instead is to link to the (not yet existing) Geospatial operators section of the RQL page.